kratos/input_output/vtk_output.cpp

```diff
@@ -105,37 +99,43 @@ void VtkOutput::WriteNodes(const ModelPart &rModelPart, std::ofstream& rFileStre
     }
 }
 
-void VtkOutput::WriteConditionsAndElements(const ModelPart &rModelPart, std::ofstream& rFileStream)
+void VtkOutput::WriteConditionsAndElements(const ModelPart& rModelPart, std::ofstream& rFileStream)
 {
-    const auto& local_mesh = rModelPart.GetCommunicator().LocalMesh();
-    if(rModelPart.NumberOfElements() > 0)
-    {
-        if(local_mesh.NumberOfConditions()>0)
+    const auto& r_local_mesh = rModelPart.GetCommunicator().LocalMesh();
+
+    int num_elements = r_local_mesh.NumberOfElements();
+    rModelPart.GetCommunicator().SumAll(num_elements);
+
+    int num_conditions = r_local_mesh.NumberOfConditions();
+    rModelPart.GetCommunicator().SumAll(num_conditions);
+
+    if (num_elements > 0) {
+        if (num_conditions > 0) {
```

Comment:
Shouldn't only the writing rank output the following ??

Reply:
what do you mean, I don't fully understand?

Reply:
I mean here you are obtaining the sum of all the num_conditions and num_elements from all the processes, so that means that the message below will be printed by all the ranks ... so wouldn't it be more clean if only the root/printing rank prints the warning ?

Reply:
ah ok
I will add a check, but in the future after jordi implemented the mpi-logger this Macro will always print on one rank only

BTW the `in (num_conditions)` is the reason the `...ONCE` macros are not working. No clue why though, I will ask Charlie in a separate issue/PR once this is merged